File: backend/src/main/java/com/cognitum/backend/dto/request/RequestSummary.java
package com.cognitum.backend.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class RequestSummary {

    private String markdown;
    @JsonProperty("max_new_tokens")
    private Integer maxNewTokens;
    private Boolean recursive;

}

File: frontend/src-tauri/src/entities/request_summary.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Serialize, Deserialize, TS)]
#[ts(export)]
pub struct RequestSummary {
    pub markdown: String,
    pub max_new_tokens: u32,
    pub recursive: bool,
}
